library/core/src/intrinsics.rs

```diff
@@ -2513,3 +2513,42 @@ where
 {
     called_in_const.call_once(arg)
 }
+
+// Wrappers around the init assertion intrinsics. Calls to these
+// are inserted in the check_maybe_uninit mir pass
+
+#[unstable(
+    feature = "maybe_uninit_checks_internals",
+    issue = "none",
+    reason = "implementation detail of panics on invalid MaybeUninit usage"
+)]
+#[rustc_allow_const_fn_unstable(const_assert_type2)]
+#[rustc_const_stable(
+    feature = "const_maybe_uninit_checks_internals",
+    since = "CURRENT_RUSTC_VERSION"
+)]
+#[cfg_attr(not(bootstrap), lang = "assert_zero_valid")]
+#[track_caller]
+#[cfg_attr(not(bootstrap), allow(unused_unsafe))]
+pub const fn assert_zero_valid_wrapper<T>() {
+    // SAFETY: Always safe to call
+    unsafe { assert_zero_valid::<T>() };
+}
```

Comment:
On the implementation side, wouldn't it be easier to directly insert calls to the intrinsics? Right now there is a library roundtrip and I cannot see a good reason for that.

Reply:
It needs to be a lang item to insert a call to it. When I tried making the intrinsic a lang item, I had a very bad time.

Reply:
Ah `Call` wants a DefId... we only care about its name when it is an intrinsic so it's a bit silly, but it's what it is I guess.